File: src/main/java/io/github/zhengchalei/common/jpa/CustomTenantResolver.java
package io.github.zhengchalei.common.jpa;

import io.quarkus.hibernate.orm.PersistenceUnitExtension;
import io.quarkus.hibernate.orm.runtime.tenant.TenantResolver;
import io.vertx.core.http.Cookie;
import io.vertx.ext.web.RoutingContext;

import javax.enterprise.context.RequestScoped;
import javax.inject.Inject;

@PersistenceUnitExtension
@RequestScoped
public class CustomTenantResolver implements TenantResolver {

    @Inject
    RoutingContext context;

    @Override
    public String getDefaultTenantId() {
        return "public";
    }

    @Override
    public String resolveTenantId() {
        Cookie cookie = context.request().getCookie("Tenant");
        if (cookie == null || cookie.getValue() == null) {
            return getDefaultTenantId();
        }
        return cookie.getValue();
    }

}

File: src/main/kotlin/io/github/zhengchalei/config/CustomTenantResolver.kt
package io.github.zhengchalei.config

import io.quarkus.hibernate.orm.PersistenceUnitExtension
import io.quarkus.hibernate.orm.runtime.tenant.TenantResolver
import io.vertx.ext.web.RoutingContext
import jakarta.enterprise.context.RequestScoped

@PersistenceUnitExtension
@RequestScoped
class CustomTenantResolver(
    private val context: RoutingContext
) : TenantResolver {

    override fun getDefaultTenantId(): String {
        return "public"
    }

    override fun resolveTenantId(): String {
        val cookie = context.request().getCookie("Tenant")
        return if (cookie == null || cookie.value == null) {
            defaultTenantId
        } else cookie.value
    }
}
